src/app/features/ledger/flows/bitcoin-tx-signing/steps/approve-bitcoin-sign-ledger-tx.tsx

```diff
@@ -20,7 +25,7 @@ export function ApproveSignLedgerBitcoinTx() {
         [
           ...getPsbtTxInputs(context.transaction as unknown as btc.Transaction).map((input, i) => [
             `Input ${i + 1}`,
-            input.witnessUtxo?.amount?.toString() + ' sats',
+            getBitcoinInputValue(isDefined(input.index) ? input.index : 0, input) + ' sats',
```

Comment:
Why do we default to 0 here?

Reply:
Thanks, I made a mistake when copying it from `useParsedInputs` where we default the value to `0`

Reply:
This is actually not working correctly now due to this change as it runs `getBitcoinInputValue` on each input and then displays the value as `0` as the last one is `undefined`

Reply:
@kyranjamie , can you please advise here? I got this working before but it seems like it was by accident as it was showing the correct amount via the default to `0`. Is it sufficient to show the amount value of the first output only? The first output being what we are sending and the second being the change we get back. 

For the transaction I am testing, `getBitcoinInputValue` is returning a `0` fallback as the `input.index` is `2`  and this doesn't match any of the outputs - there are only 2 outputs. 

In `getBitcoinInputValue` the logic is: 
```
export function getBitcoinInputValue(index: number, input: btc.TransactionInput) {
  if (isDefined(input.witnessUtxo)) return Number(input.witnessUtxo.amount);
  if (isDefined(input.nonWitnessUtxo)) return Number(input.nonWitnessUtxo.outputs[index]?.amount);
  return 0;
}
```

So in this case:
- `input.nonWitnessUtxo` is defined 
- it tries to return `input.nonWitnessUtxo.outputs[index]?.amount`
- I am passing an `input.index` of `2` but this doesn't match any of the outputs

The raw transaction inputs are: 
```
 "inputs": [
      {
        "nonWitnessUtxo": {
          "version": 2,
          "segwitFlag": true,
          "inputs": [
            {
              "txid": "01d0c5d6797503708da59613a0f5ccc2d54c6d20f64ccf3724370795cc09443f",
              "index": 2,
              "finalScriptSig": "",
              "sequence": 0
            }
          ],
          "outputs": [
            {
              "amount": "250000",
              "script": "00149030ae5915740aca30c76812e2f052fb3388d485"
            },
            {
              "amount": "1327380",
              "script": "0014aa590a637e0490e383af48e953e5ce35ff4baec4"
            }
          ],
```

Reply:
It could be this logic is wrong. All the references to index are confusing as there is ambiguity to what it refers. We've encountered bug here before, that I thought'd been fixed.

A transaction input spends an existing transaction's output. Both the new input being spent, and the existing output have an index. Nothing guarantees the input will be at the same index as the output it spends.

Here, it does look wrong me me, because we're using the input's index to refer back to the `nonWitnessUtxo` to find the value.

cc/ @fbwoolf for visibility here. Do you recall the bug we had relating to this? Isn't it odd that this function receives an input, and the index separately that's part of the input?

Reply:
Scratch that, `input.index` is the identifier to the output index (not it's own position in the new tx).

Reply:
See https://github.com/leather-wallet/extension/pull/5363

Reply:
OK thanks. I won't use the function in this case and will instead take the amount of the first amount for `nonWitnessUtxo`. 

AKA `nonWitnessUtxo.outputs[0]`. Is that reliable or is there a case where there could be several outputs (more than two)

Reply:
I wouldn't default to the first one, this could show completely the wrong information.

> Correct, the index here isn't referring to this txs outputs index, it is trying to get the value of the input.

@fbwoolf The value of the input is defined inside the output it's spending right?

> I am passing an input.index of 2 but this doesn't match any of the outputs

@pete-watters curious how this is possible. Doesn't that mean that it's an invalid input?

Would be great if we can find a way to fix this function so it's accurate.

Reply:
> @fbwoolf The value of the input is defined inside the output it's spending right?
> 
That is correct. It should be looking at the output values from the UTXO data provided.

Reply:
@pete-watters instead of looking at the raw transaction, are you logging in the console what `getPsbtTxInputs` returns using the `getInput` method from btc-signer lib? That is what is being used here, right?

Reply:
Was thinking on this, instead of returning `0` here, shouldn't we flag it as an invalid input? The app could def be sending us an invalid input in the PSBT.

Reply:
Thanks @fbwoolf and @kyranjamie 👏 

Fara, you were right, I was confused again and I was passing the `input.index` of the `nonWitnessUtxo.inputs` rather than the output of `getPsbtInputs`. That is set to `2` so that explains what happened. 

![Screenshot 2024-05-14 at 06 50 49](https://github.com/leather-wallet/extension/assets/2938440/80b5c3df-120d-4a4f-859f-f21c0fff7978)

Now with Kyrans [refactor](https://github.com/leather-wallet/extension/pull/5363) it's working without passing the index. 

I was doing this:
```
...getPsbtTxInputs(context.transaction as unknown as btc.Transaction).map((input, i) => [
            `Input ${i + 1}`,
            input.index && `${getBitcoinInputValue(input.index, input)} sats`,
          ]),
```

But it should have been:
```
const psbtOutputIndex = getPsbtTxInputs(context.transaction as unknown as btc.Transaction).input.index; 

...

...getPsbtTxInputs(context.transaction as unknown as btc.Transaction).map((input, i) => [
            `Input ${i + 1}`,
            `${getBitcoinInputValue(psbtOutputIndex, input)} sats`,
          ]),
```

Thanks to both of you for your patience helping here